osu.Game/Database/BeatmapMetadata.cs

```diff
@@ -9,8 +9,9 @@ public class BeatmapMetadata
     {
         [PrimaryKey, AutoIncrement]
         public int ID { get; set; }
-        
-        public int BeatmapSetID { get; set; }
+
+        public int BeatmapSetID { get; set; } = -1;

```

Comment:
same here (nullable). will also make sense after the rename to Online... happens, since right now this is referring to the PK ID but has the same name of the non-PK ID.